client/sites/overview/components/test/hosting-overview.tsx

```diff
@@ -0,0 +1,330 @@
+/**
+ * @jest-environment jsdom
+ */
+import { MigrationStatus } from '@automattic/data-stores';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { render, waitFor } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import { createReduxStore } from 'calypso/state';
+import { SITE_REQUEST, SITES_REQUEST, SELECTED_SITE_SET } from 'calypso/state/action-types';
+import { setStore } from 'calypso/state/redux-store';
+import * as siteActions from 'calypso/state/sites/actions';
+import HostingOverview from '../hosting-overview';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { UnknownAction } from 'redux';
+
+// Store original location so we can restore it after all tests.
+const originalLocation = window.location;
+
+type LocalRenderWithProviderOptions = {
+	initialState?: Record< string, unknown >;
+	additionalActions?: UnknownAction[];
+};
+const localRenderWithProvider = (
+	element,
+	{ initialState, additionalActions = [] }: LocalRenderWithProviderOptions
+) => {
+	const store = createReduxStore( initialState ?? {} );
+
+	setStore( store );
+
+	additionalActions.forEach( ( action ) => {
+		store.dispatch( action );
+	} );
+
+	const queryClient = new QueryClient();
+	return render(
+		<QueryClientProvider client={ queryClient }>
+			<Provider store={ store }>{ element }</Provider>
+		</QueryClientProvider>
+	);
+};
```

Comment:
Note that I've created a dedicated PR to discuss hoisting more of this code into a general-purpose test helper in https://github.com/Automattic/wp-calypso/pull/101587.